src/main/java/views/create_new_month/NewMonthV.java

```diff
@@ -0,0 +1,151 @@
+
+        // Add all components to layout
+        layout.add(yearMonthPanel);
+        layout.add(buttonPanel);
+
+        // Set up frame
+        layout.setBorder(BorderFactory.createEmptyBorder(50, 60, 50, 60));
+        frame.setContentPane(layout);
+        frame.pack();
+        frame.setDefaultCloseOperation(JFrame.HIDE_ON_CLOSE);
+        frame.setVisible(true);
+    }
+
+    /**
+     * React to drop down selection and button click that result in ActionEvent.
+     * Code inspired from <a href="https://youtu.be/Kmgo00avvEw">here</a>.
+     * @param evt the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent evt) {
+        if (evt.getSource()==submitButton){
+            // Check that user has inputted
+            if (Objects.equals(year.getText(), "")||Objects.equals(budget.getText(), "")){
+                JOptionPane.showMessageDialog(frame,"Please fill in text fields.");
+            } else {
+                //call helper method when there is input
+                getOutput();
+            }
+        } else if (evt.getSource()==month) {
+            // Get integer representation of selected Month
+            this.selectedMonth = month.getSelectedIndex();
+        }
+    }
+
+    /**
+     * Helper method to all controller method when user input is valid.
+     */
+    private void getOutput(){
+        try {
+            // Change user input to required number types
+            int yearInt = Integer.parseInt(year.getText());
+            int monthID = (yearInt * 100) + selectedMonth;
+            double budgetValue = Double.parseDouble(budget.getText());
+
+            if (yearInt < 1900 || yearInt > 2100){
+                JOptionPane.showMessageDialog(frame, "Input a valid year.");
+            } else if (selectedMonth == 0) {
+                JOptionPane.showMessageDialog(frame,"Please select a Month.");
+            } else if (budgetValue <= 0){
+                JOptionPane.showMessageDialog(frame, "Budget must be more than 0.");
+            } else {
+                // Create new MonthlyStorage and get output
+                NewMonthOD output = controller.getOutput(session, monthID, budgetValue);
+                if (output.isSuccessful()){
+                    loadMonthMenu(session,monthID,"Month created successfully.");
+                } else {
+                    JOptionPane.showMessageDialog(frame, output.getWarning());
+                }
+            }
+        } catch (NumberFormatException e){
+                JOptionPane.showMessageDialog(frame,"Please input valid numbers.");
+        }
+    }
+
+    /**
+     * Load Month Menu and notify user.
+     * @param session the SessionStorage holding the required MonthlyStorage
+     * @param monthID the monthID of the required MonthlyStorage
+     * @param message notify user that Month Menu is updated
+     */
+    @Override
+    public void loadMonthMenu(SessionStorage session, int monthID, String message) {
```

Comment:
I can't recall if this was something said in CSC148 or just a preference, but I put all the private methods below the public ones. I can't recall if this was an actual thing or not, but I think it is a bit easier to read and follow.

Reply:
I'll hold on and change this later, as I might change loadMonthMenu() to private later. But thanks for telling me that, I didn't know!

Reply:
After discussing about the interface with Ari, I decided that I will keep the interface, so I've moved loadMonthMenu() before the private method!

Reply:
Sounds good!